### PR TITLE
Added matrix profile using stumpy 

### PR DESCRIPTION
**Pull request by utsavcoding**
matrix_profile.py with soft dependency check for stumpy added

#### Reference Issues/PRs

https://github.com/alan-turing-institute/sktime/issues/450

#### What does this implement/fix? Explain your changes.

Interface and wrap stumpy.stump in series-to-series transformer.

#### Does your contribution introduce a new dependency? If yes, which one?

No hard dependency, but stumpy is introduced as soft dependency

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/master/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/master/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Comment by mloning** [comment]
As a next step, take a look at the `CONTRIBUTING.md` to find out where you have to add stumpy to our dependency specification files.

**Comment by utsavcoding** [comment]
> As a next step, take a look at the `CONTRIBUTING.md` to find out where you have to add stumpy to our dependency specification files.

I checked out `CONTRIBUTING.md`, but I think those are relevant to hard dependency. Do I need to add something for the soft dependency?

**Comment by mloning** [comment]
@utsavcoding in order to get the unit tests to pass you need to add it in `build_tools/requirements.txt` and in `setup.py` extra requirements. If you want to add a tutorial notebook later on, you also have to add it in `.binder/requirements.txt`.

File: sktime/transformers/series/matrix_profile.py [hunk 1]
```diff
@@ -1,4 +1,56 @@
...
+            of the desired length in each time series.
+        """
+        self.check_is_fitted()
+        X = check_series(X, enforce_univariate=True)
+        n_instances = X.shape[0]
+        Xt = pd.DataFrame([_stumpy_series(X[i], self.m) for i in range(n_instances)])
+        return Xt
+
+
+df = pd.read_csv("/home/utsav/Downloads/timeseries.csv")
```
Comment (utsavcoding) [inline sktime/transformers/series/matrix_profile.py:53]: Ignore these lines of code, this is just for testing purpose.

File: sktime/transformers/series/matrix_profile.py [hunk 2]
```diff
@@ -1,4 +1,56 @@
...
+           Time series dataset.
+
+        Returns
+        -------
+        Xt: pandas.DataFrame
+            Dataframe with the same number of rows as the input.
+            The number of columns equals the number of subsequences
+            of the desired length in each time series.
+        """
+        self.check_is_fitted()
```
Comment (utsavcoding) [inline sktime/transformers/series/matrix_profile.py:46]: This instance of {self.__class__.__name__} has not "
sktime.exceptions.NotFittedError: This instance of MatrixProfileTransformer has not been fitted yet; please call `fit` first.

I am getting the above error in this line.

**Comment by utsavcoding** [comment]
> @utsavcoding in order to get the unit tests to pass you need to add it in `build_tools/requirements.txt` and in `setup.py` extra requirements. If you want to add a tutorial notebook later on, you also have to add it in `.binder/requirements.txt`.

**Review by mloning — COMMENTED** [review]
Hi @utsavcoding,  in order to get the unit tests to pass you need to add it in `build_tools/requirements.txt` and in `setup.py` extra requirements. If you want to add a tutorial notebook later on, you also have to add it in `.binder/requirements.txt`.

Let me know if you need help with that, I don't see the error that you see at the moment because stumpy isn't installed because it is not specified in the requirements.

**Comment by mloning** [comment]
FYI we had some CI issues, but they're now fixed on master, so you need to update your branch with the latest changes on master.

**Comment by utsavcoding** [comment]
@mloning I have synced my local master branch with the latest updates. And I have added the dependency in as per mentioned above, but I am still facing a key error issue due to which 2 of the pytest(test_all_estimators.py, test_all_transformers.py) are failing.

**Comment by mloning** [comment]
FYI your master was still out-of-sync (see my latest commit), I'm going to take a look over the next few days!

**Comment by utsavcoding** [comment]
@mloning I have pushed in the changes with default window_length=3 for matrix profile. Can you please review it.

**Comment by utsavcoding** [comment]
@mloning Added configuration for stumpy in check_soft_dependencies.py(failure of azure build in github).

**Review by mloning — APPROVED** [review]
Nice work @utsavcoding, all tests pass, everything looks good to me! I'll leave it open a bit longer in case others want to comment before we merge.

**Comment by mloning** [comment]
Thanks again @utsavcoding - great work!

**Pull request merged**